File: velocidade.py
def calcular_aceleracao(velocidade_inicial, velocidade_final, tempo):
  """
  Calcula a aceleração (taxa de variação de velocidade).

  Argumentos:
    velocidade_inicial (float): A velocidade no início do intervalo de tempo.
    velocidade_final (float): A velocidade no final do intervalo de tempo.
    tempo (float): O tempo decorrido para a variação da velocidade.

  Retorna:
    float: A aceleração calculada.
  """
  aceleracao = (velocidade_final - velocidade_inicial) / tempo
  return aceleracao

def main():
  """
  Função principal para executar o programa.
  """
  print("--- Calculadora de Aceleração ---")
  print("A aceleração é a taxa de variação de velocidade ao longo do tempo.")

  try:
    # Solicita a entrada do usuário
    velocidade_inicial = float(input("Digite a velocidade inicial (em m/s): "))
    velocidade_final = float(input("Digite a velocidade final (em m/s): "))
    tempo = float(input("Digite o tempo decorrido (em segundos): "))

    # Verifica se o tempo é zero para evitar divisão por zero
    if tempo == 0:
      print("Erro: O tempo decorrido não pode ser zero.")
      return

    # Chama a função para calcular a aceleração
    aceleracao_calculada = calcular_aceleracao(velocidade_inicial, velocidade_final, tempo)

    # Exibe o resultado
    print(f"\nResultados:")
    print(f"Velocidade Inicial: {velocidade_inicial} m/s")
    print(f"Velocidade Final: {velocidade_final} m/s")
    print(f"Tempo Decorrido: {tempo} s")
    print(f"A Aceleração é: {aceleracao_calculada:.2f} m/s²")
    print("---------------------------------")

  except ValueError:
    print("Entrada inválida. Por favor, insira apenas números.")
  except Exception as e:
    print(f"Ocorreu um erro: {e}")

# Executa a função principal quando o script é rodado
if __name__ == "__main__":
  main()

File: velocidade .c
#include <stdio.h>

// Função para calcular a aceleração
float calcular_aceleracao(float velocidade_inicial, float velocidade_final, float tempo) {
    // A fórmula da aceleração é: a = (velocidade_final - velocidade_inicial) / tempo
    return (velocidade_final - velocidade_inicial) / tempo;
}

int main() {
    // Declaração das variáveis
    float velocidade_inicial;
    float velocidade_final;
    float tempo;
    float aceleracao;

    printf("--- Calculadora de Aceleracao ---\n");
    printf("A aceleracao e a taxa de variacao de velocidade ao longo do tempo.\n");

    // Solicita a entrada do usuário
    printf("Digite a velocidade inicial (em m/s): ");
    scanf("%f", &velocidade_inicial);

    printf("Digite a velocidade final (em m/s): ");
    scanf("%f", &velocidade_final);

    printf("Digite o tempo decorrido (em segundos): ");
    scanf("%f", &tempo);

    // Verifica se o tempo e zero para evitar a divisao por zero
    if (tempo == 0) {
        printf("Erro: O tempo decorrido nao pode ser zero.\n");
        return 1; // Retorna 1 para indicar um erro
    }

    // Chama a funcao para calcular a aceleracao
    aceleracao = calcular_aceleracao(velocidade_inicial, velocidade_final, tempo);

    // Exibe o resultado
    printf("\nResultados:\n");
    printf("Velocidade Inicial: %.2f m/s\n", velocidade_inicial);
    printf("Velocidade Final: %.2f m/s\n", velocidade_final);
    printf("Tempo Decorrido: %.2f s\n", tempo);
    printf("A Aceleracao e: %.2f m/s²\n", aceleracao);
    printf("----------------------------------\n");

    return 0; // Retorna 0 para indicar que o programa foi executado com sucesso
}